src/mlpack/methods/ann/loss_functions/multilabel_softmargin_loss_impl.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file methods/ann/loss_functions/multilabel_softmargin_loss_impl.hpp
+ * @author Anjishnu Mukherjee
+ *
+ * Implementation of the Multi Label Soft Margin Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_IMPL_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_IMPL_HPP
+
+// In case it hasn't been included.
+#include "multilabel_softmargin_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+MultiLabelSoftMarginLoss<InputDataType, OutputDataType>::
+MultiLabelSoftMarginLoss(
+    const bool reduction,
+    const arma::rowvec& weights) :
+    reduction(reduction),
+    weighted(false)
+{
+  if (weights.n_elem)
+  {
+    classWeights = weights;
+    weighted = true;
+  }
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+typename InputType::elem_type
+MultiLabelSoftMarginLoss<InputDataType, OutputDataType>::Forward(
+    const InputType& input, const TargetType& target)
+{
+  if (!weighted)
+    classWeights.ones(1, input.n_cols);
```

Comment:
Hmm, The weights will be initialised every time in forward pass here, You should set weighted to true here to prevent that from happening. As shown [here](https://github.com/mlpack/mlpack/pull/2345#issuecomment-620726423), we use reset to prevent weights from being initialized every time i.e. If user has not passed weights they will be initialized once in forward pass.

Reply:
Agreed! This was something I missed completely. Corrected in the commit 77463d8a4a260c697a6d4705ca30d38a1503c3ef.
Thanks to both zoq and you for pointing this out.